frontend/apps/reader/modules/readerview.lua

```diff
@@ -815,11 +815,29 @@ end
 
 function ReaderView:onSetRotationMode(rotation)
     if rotation ~= nil then
-        if rotation == Screen:getRotationMode() then
+        if rotation == Screen:getRotationMode() then -- no change
             return true
         end
+
+        local requested_screen_mode -- landscape or portrait
+        if rotation == Screen.DEVICE_ROTATED_UPRIGHT or rotation == Screen.ICE_ROTATED_UPSIDE_DOWN then
+            requested_screen_mode = "portrait"
+        else
+            requested_screen_mode = "landscape"
+        end
+
+        local old_rotation_mode = Screen:getRotationMode()
+        local old_screen_mode = Screen:getScreenMode()
+        if rotation and rotation ~= old_rotation_mode and requested_screen_mode == old_screen_mode then
+            -- Cheaper than a full SetRotationMode event, as we don't need to re-layout anything.
+            Screen:setRotationMode(rotation)
+            UIManager:onRotation()
+            return true
```

Comment:
You also no longer get a notification by aborting early.

Reply:
We have notification for everything happening from the bottom menu, so I wouldn't necessarily want to single that one out, even if it sounds fairly self-evident ;).